### ISSUE TITLE
KB5027303 breaks Open-Shell - and every other? - Explorer extension

### ISSUE THREAD

**Issue by VulcanTourist** [Open-Shell/Open-Shell-Menu#1580]
### Describe the bug

After Windows update KB5027303, the behavior and UI of Explorer is changed and seems to no longer support extensions. Explorer's View menu no longer includes any options to manage the behavior of extensions, and they appear to have been disabled, all of them, not just Open-Shell.

Is there a means to work around this, or is the Open-Shell Explorer extension effectively dead for Windows 11 now?

### Area of issue

Windows Explorer

### To reproduce

1. Spawn an instance of Explorer, note that Open-Shell UI extension that existed prior to KB5027303 is gone;
2. navigate to View menu, intending to do what was done many times before, to reactivate a disabled extension;
3. note with dismay that former menu options for managing extension are GONE, with no alternative.

### Expected behavior

No change in UI extension availability.

### Open-Shell version

4.4.189

### Windows version

Windows 10 22H2 build 22621.1928 (post-KB5027303)

### Additional context

Open-Shell is not the only Explorer extension affected by this update and new Windows build; StExBar has also been eliminated, among others.

**Comment by ge0rdi** [comment]
I have Win11 22621.1928 (fully updated) on VM and `Open-Shell` toolbar is visible in Explorer:

![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/2453094/653baab1-0fd9-4341-b527-03d6ace0eb5d)

KB5027303 is installed:
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/2453094/7fd8f587-0863-4386-9637-a965fa9fa35b)

**Comment by VulcanTourist** [comment]
> I have Win11 22621.1928 (fully updated) on VM and `Open-Shell` toolbar is visible in Explorer:
> 
> ![image](https://user-images.githubusercontent.com/2453094/251366831-653baab1-0fd9-4341-b527-03d6ace0eb5d.png)
> 
> KB5027303 is installed: ![image](https://user-images.githubusercontent.com/2453094/251366945-7fd8f587-0863-4386-9637-a965fa9fa35b.png)

While you're lucky to happen to still have it appearing, if you wanted to disable it from appearing in Explorer, where are the Explorer menu items to do that?  They no longer exist under View as they did prior to this update.  Because of that, I can't even try to re-enable it (and other missing UI extensions) as I formerly have whenever they spontaneously disappeared (and they have numerous times over the years).

**Comment by daknut** [comment]
Update KB5027303 did it for me with Open Shell, too. It left all the files and the program exactly where they were but W11 refused to allow the program to run on my and four other laptops. I don't use the IE UI but I uninstalled Open Shell and am unable to use it anymore.

**Comment by Bluscream** [comment]
Same here, open shell does not open anymore. Is there a manual way to trigger the menu from cmd or shortcut?

**Comment by daknut** [comment]
> Is there a manual way to trigger the menu from cmd or shortcut?

I didn't try cmd but I couldn't work out a way to override the W11 Start Menu. What I *did* find was a 'startscreen' file at C:\Users\[User name]\AppData\Roaming\Microsoft\Windows\Start Menu which appeared after the upgrade.

I'm guessing that it's responsible in some way. I gave up trying to make Open Shell work and have now deleted it. Along with the four other laptops it was being run on.

Some users have had no problem at all - which begs the question whether it might be regional. We're in the UK.

**Comment by VulcanTourist** [comment]
> Update KB5027303 did it for me with Open Shell, too. It left all the files and the program exactly where they were but W11 refused to allow the program to run on my and four other laptops. I don't use the IE UI but I uninstalled Open Shell and am unable to use it anymore.

> Same here, open shell does not open anymore. Is there a manual way to trigger the menu from cmd or shortcut?
> 
> List of installed programs and updates:
> 
> [programs.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/12061914/programs.txt) [uninstall.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/12061918/uninstall.txt)

> > Is there a manual way to trigger the menu from cmd or shortcut?
> 
> I didn't try cmd but I couldn't work out a way to override the W11 Start Menu. What I _did_ find was a 'startscreen' file at C:\Users[User name]\AppData\Roaming\Microsoft\Windows\Start Menu which appeared after the upgrade.
> 
> I'm guessing that it's responsible in some way. I gave up trying to make Open Shell work and have now deleted it. Along with the four other laptops it was being run on.
> 
> Some users have had no problem at all - which begs the question whether it might be regional. We're in the UK.

What you're both describing is a completely unrelated issue to the one that I've reported here, and in the process you're diluting all our efforts.  Please create a distinct Issue.  What I have described here is limited ***only*** to Open-Shell's toolbar extension for Windows Explorer.  KB5027303 ***did not*** entirely disable Open-Shell or its replacement Start menu for me, as you are both reporting.

What it did do is completely remove the *View* menu options from Windows Explorer that allow managing Explorer UI (toolbar) extensions; those options are now just *GONE*, suggesting that Microsoft either has already or is planning to fully disable the extension API for Explorer.  The writing is on the wall... can anything be done to rescue the Open-Shell toolbar's functionality?

**Comment by daknut** [comment]
> What you're both describing is a completely unrelated issue to the one that I've reported here, and in the process you're diluting all our efforts. Please create a distinct Issue.

I did report the issue - it was closed and removed.

**Comment by VulcanTourist** [comment]
> > What you're both describing is a completely unrelated issue to the one that I've reported here, and in the process you're diluting all our efforts. Please create a distinct Issue.
> 
> I did report the issue - it was closed and removed.

... and so you think it's just fine to sneak into another issue and hijack it for your purpose?  How very Machiavellian of you.

**Comment by daknut** [comment]
> ... and so you think it's just fine to sneak into another issue and hijack it for your purpose? How very Machiavellian of you.

Not at all. You noted that the same Windows Update that caused the problem for us was the one that did it for you. That's what I was responding to.

But please try to make the web a nicer place to be by being kind. Everyone benefits

**Comment by VulcanTourist** [comment]
> > ... and so you think it's just fine to sneak into another issue and hijack it for your purpose? How very Machiavellian of you.
> 
> Not at all. You noted that the same Windows Update that caused the problem for us was the one that did it for you. That's what I was responding to.
> 
> But please try to make the web a nicer place to be by being kind. Everyone benefits

That Windows Update ***is not itself*** the Issue being reported, and again you're simplifying for your own selfish motive.  The Update has had - or can have - multiple effects on Open-Shell.  You tried to report one such effect, and for whatever reason the project admins decided to close it.  What I'm reporting here is ***another*** distinct effect of that same Update:

-- *KB5027303 breaks Open-Shell Explorer extension*

Discussion of your separate issue and Update effect does not belong here.

**Comment by Bluscream** [comment]
I have the same issue aswell, so the chance that he's "trolling" is zero. Invalidating such a serious issue as "program does not start anymore" seems like a linux move, not used to that from anything else. I would've created a seperate issue, but after hearing it'll just get closed anyway there's no reason to. And the fact that our issue is handled this way makes me seriously question wether i would want to continue using open shell, even if it **would word** in the first place.

**Comment by ge0rdi** [comment]
> after hearing it'll just get closed anyway there's no reason to

The issue @daknut is talking about is #1571.
It was closed because he didn't seem to be willing to cooperate on investigating it.

If you are experiencing the same issue after the same Windows update then please add comment there (I guess it should be possible even for closed issues). You can even create new issue and link the other one so that it is clear more users have the same issue.

But, please, lets not discuss unrelated things here.
This issue is about Explorer extension as it was pointed out already.

**Comment by VulcanTourist** [comment]
4.4.190 does not correct this issue.  I still have toolbar extension missing, as well as that for other Explorer extensions, and the menu items for dealing with them are GONE.
![View menu](https://github.com/Open-Shell/Open-Shell-Menu/assets/6064894/7c769004-ef0d-49e3-a135-bdb77fe64176)
![View - Show menu](https://github.com/Open-Shell/Open-Shell-Menu/assets/6064894/1cd0e7ef-db37-45e4-b55a-353fcc306bd0)
![extra menu](https://github.com/Open-Shell/Open-Shell-Menu/assets/6064894/4b96fb55-444c-42d0-83cc-717da8353830)

**Comment by VulcanTourist** [comment]
> I'll mention it here if there is anything to test/try.

Please do, though I'm not hopeful; the missing menu items suggests that they're planning to completely remove extension support and started with removing the menu UI for it.  I REALLY have been dependent upon both the toolbar extensions that I was using, and now there's at least one thing I cannot do any longer because nothing replicates what it did.
